projects/etos_suite_runner/src/etos_suite_runner/esr.py

```diff
@@ -64,21 +66,23 @@ def __init__(self) -> None:
             int(os.getenv("ESR_WAIT_FOR_ENVIRONMENT_TIMEOUT")),
         )
 
-    def _request_environment(self, ids: list[str]) -> None:
+    def _request_environment(self, ids: list[str], otel_context_carrier: dict) -> None:
         """Request an environment from the environment provider.
 
         :param ids: Generated suite runner IDs used to correlate environments and the suite
                     runners.
+        :param otel_context_carrier: a dict carrying current OpenTelemetry context.
         """
         span_name = "request_environment"
-        suite_context = get_current_context()
+        suite_context = TraceContextTextMapPropagator().extract(carrier=otel_context_carrier)
+        opentelemetry.context.attach(suite_context)
```

Comment:
The token that is received from `attach` should be detached in a finally block.
This comment applies to all instances where context attachment is done